src/components/MainContainer.js

```diff
@@ -42,7 +43,7 @@ class MainContainer extends Component {
 
             this.setState({dataset, actuals, indicators});
         } catch (e) {
-            console.log('Fetch was problematic: ' + e.message)
+            this.setState({fetchErrors: e.message});
         } finally {
             this.setState({dataLoaded: true});
```

Comment:
@BenjiLee I'm wondering if it's better just to add a catch inside the finally, like so:

```
finally {
   if (Object.keys(dataset).length > 0 && 
            Object.keys(actuals).length > 0 && 
            Object.keys(indicators).length > 0) {

            this.setState({dataLoaded: true});
}
```

Reply:
That way `dataLoaded` is only flipped on if we know all the data files exists.

Reply:
That can also work.